### PR TITLE
doc: Document Python dependency repos and search

### PR DESCRIPTION
**Pull request by tsenart**
## Test plan

It's docs, CI has link checks.

### CODE REVIEW

File: doc/code_search/how-to/dependencies_search.md [hunk 1]
```diff
@@ -29,20 +29,26 @@ Search only Go dependencies:
...
-[Go](../../integration/go.md)   | lsif-go uploads           | ❌     | ❌
-[JVM](../../integration/jvm.md) | `gradle.lockfile`         | ❌     | ❌
-[JVM](../../integration/jvm.md) | `pom.xml`                 | ❌     | ❌
-Python                          | `poetry.lock`             | ❌     | ❌
+Kind                                  | How                       | Direct | Transitive
+-------------------------------       |-------------------------- |------- | ----------
+[npm](../../integration/npm.md)       | `package-lock.json`       | ✅     | ✅
+[npm](../../integration/npm.md)       | lsif-typescript uploads   | ✅     | ✅
+[npm](../../integration/npm.md)       | `yarn.lock`               | ✅     | ✅
+[Python](../../integration/python.md) | `poetry.lock`             | ✅     | ✅
```
Comment (mpimenov) [inline doc/code_search/how-to/dependencies_search.md:47]: I think it's safe to add Pipfile.lock here too, I expect the corresponding PR to be merged today (it has all the checks passed)

File: CHANGELOG.md [hunk 2]
```diff
@@ -26,6 +26,7 @@ All notable changes to Sourcegraph are documented in this file.
 - Search-based code intelligence now recognizes local variables in Python, Java, JavaScript, TypeScript, C/C++, C#, Go, and Ruby. [#33689](https://github.com/sourcegraph/sourcegraph/pull/33689)
 - GraphQL API: Added support for async external service deletion. This should be used to delete an external service which cannot be deleted within 75 seconds timeout due to a large number of repos. Usage: add `async` boolean field to `deleteExternalService` mutation. Example: `mutation deleteExternalService(externalService: "id", async: true) { alwaysNil }`
 - [search.largeFiles](https://docs.sourcegraph.com/admin/config/site_config#search-largeFiles) now supports recursive globs. For example it is now possible to specify a pattern like `**/*.lock` to match a lock file anywhere in a repository. [#35411](https://github.com/sourcegraph/sourcegraph/pull/35411)
+- The `repo:dependencies(...)` predicate can now search through the [Python dependencies of your repositories](https://docs.sourcegraph.com/code_search/how-to/dependencies_search). [#32659](https://github.com/sourcegraph/sourcegraph/issues/32659)
```
Comment (mpimenov) [inline CHANGELOG.md:29]: nit: maybe include ```(short form `r:deps(...))` ``` for those who only read the changelog and don't read docs

File: doc/code_search/how-to/dependencies_search.md (hunk 1 above)
Comment (tsenart) [reply doc/code_search/how-to/dependencies_search.md:47]: Great! Will update.

Comment (mpimenov) [reply doc/code_search/how-to/dependencies_search.md:47]: The PR is merged now.

File: CHANGELOG.md (hunk 2 above)
Comment (tsenart) [reply CHANGELOG.md:29]: Made it `repo:deps()` since I think many users don't know that `r:` is an alias of `repo:`

**Review by stefanhengl — APPROVED** [review]

File: doc/integration/python.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+You can use Sourcegraph with Python packages from any Python package mirror, including open source code from pypi.org or a private mirror such as [Nexus](https://www.sonatype.com/products/nexus-repository).
+This integration makes it possible to search and navigate through the source code of published Python packages (for example, [`numpy@v1.19.5`](https://sourcegraph.com/python/numpy@v1.19.5)).
+
+Feature | Supported?
+------- | ----------
+[Repository syncing](#repository-syncing) | ✅
+[Credentials](#credentials) | ✅
+[Rate limiting](#rate-limiting) | ✅
+Repository permissions | ❌
```
Comment (mpimenov) [inline doc/integration/python.md:11]: Oh I meant changing the hashtag part and not removing the link completely

**Review by mpimenov — COMMENTED** [review]
LGTM

**Pull request merged**